samplesheets/views_api.py

```diff
@@ -137,6 +147,288 @@ def post(self, request, *args, **kwargs):
+            sodar_uuid=self.kwargs.get('irodsdatarequest')
+        )
+
+    def delete(self, request, *args, **kwargs):
+        """DELETE request for deleting an iRODS data request"""
+        obj = self.get_object()
+        if not self.check_irods_permissions(request, obj):
+            raise PermissionDenied('Insufficient permissions')
+
+        ex_msg = 'Deleting' + IRODS_EX_MSG + ':'
+        irods_request = IrodsDataRequest.objects.filter(
+            sodar_uuid=self.kwargs.get('irodsdatarequest')
+        ).first()
+        try:
+            irods_request.delete()
+        except Exception as ex:
+            raise APIException('{} {}'.format(ex_msg, ex))
+
+        # Add timeline event
+        self.add_tl_delete(irods_request)
+        # Handle project alerts
+        self.handle_alerts_deactivate(irods_request)
+        return Response(
+            {'detail': 'iRODS data request deleted'},
+            status=status.HTTP_200_OK,
+        )
+
+
+class IrodsRequestAcceptAPIView(
+    IrodsRequestModifyMixin, SODARAPIBaseProjectMixin, APIView
+):
+    """
+    Accept an iRODS data request for a project.
+
+    **URL:** ``/samplesheets/api/irods/request/accept/{IrodsDataRequest.sodar_uuid}``
+
+    **Methods:** ``POST``
+    """
+
+    http_method_names = ['post']
+    permission_required = 'samplesheets.manage_sheet'
+
+    def post(self, request, *args, **kwargs):
+        """POST request for accepting an iRODS data request"""
+        timeline = get_backend_api('timeline_backend')
+        taskflow = get_backend_api('taskflow')
+        app_alerts = get_backend_api('appalerts_backend')
+        project = self.get_project()
+        ex_msg = 'Accepting' + IRODS_EX_MSG + ':'
+
+        irods_request = IrodsDataRequest.objects.filter(
+            sodar_uuid=self.kwargs.get('irodsdatarequest')
+        ).first()
+
+        try:
+            self.accept_request(
+                irods_request,
+                project,
+                self.request,
+                timeline=timeline,
+                taskflow=taskflow,
+                app_alerts=app_alerts,
+            )
+        except Exception as ex:
+            raise APIException('{} {}'.format(ex_msg, ex))
+        return Response(
+            {
+                'detail': 'iRODS data request accepted',
+                'path': irods_request.path,
+            },
+            status=status.HTTP_200_OK,
+        )
+
+
+class IrodsRequestRejectAPIView(
+    IrodsRequestModifyMixin, SODARAPIBaseProjectMixin, APIView
+):
+    """
+    Reject an iRODS data request for a project.
+
+    **URL:** ``/samplesheets/api/irods/request/reject/{IrodsDataRequest.sodar_uuid}``
+
+    **Methods:** ``GET``
+    """
+
+    http_method_names = ['get']
+    permission_required = 'samplesheets.manage_sheet'
+
+    def get(self, request, *args, **kwargs):
+        """POST request for rejecting an iRODS data request"""
+        ex_msg = 'Rejecting' + IRODS_EX_MSG + ':'
```

Comment:
Here we can also just form the message when raising `APIException`, since it's only used if an exception is met.